File: Csharp/Show/Show/ConcreteDecoratorB.cs
namespace Show
{
    class ConcreteDecoratorB: Decorator
    {
        public ConcreteDecoratorB(Component component) : base(component)
        {
        }
        
        public override string Operation()
        {
            return $"ConcreteDecoratorB({base.Operation()})";
        }

    }
}

File: JavaScript/Structural/Decorator/Example1/ConcreteDecoratorB.js
// ConcreteDecoratorB.js

const Decorator = require('./Decorator');

// ConcreteDecoratorB adds its own behavior
class ConcreteDecoratorB extends Decorator {
    operation() {
        // wrap with even more behavior
        return `ConcreteDecoratorB(${super.operation()})`;
    }
}

module.exports = ConcreteDecoratorB;